### ISSUE TITLE
Strange `invmod` results for unsigned types

### ISSUE THREAD

**Issue by TotalVerb** [JuliaLang/julia#17595]
This behaviour seems undesirable to me

```
julia> invmod(0x2, 0x3)
0xff
```

Maybe arguments to this function should be passed through `signed`.

**Comment by eschnett** [comment]
Why is the behaviour undesirable? Would you have expected a smaller result, e.g. `0x2`?

**Comment by TotalVerb** [comment]
I would have expected a result between `0x1` and `0x2`, yes. Note also that

```julia
julia> mod(0xff * 0x2, 0x3)
0x02
```

which means that this result is not an inverse at all.

**Comment by eschnett** [comment]
Indeed. I didn't catch this; it seems I mentally made (maybe) the same error as the `mod` algorithm.

I've taken the liberty of updating the title, since the result is not merely "strange" but plain "wrong".

Instead of converting the input to a signed type, I'd suggest keeping things unsigned, but correcting the error, possibly by preventing a `-1` -> `typemax` wrap-around.

**Comment by eschnett** [comment]
I have a patch; will submit PR soon.
